app/assets/javascripts/shopify_app/itp_polyfill.js

```diff
@@ -0,0 +1,8 @@
+function shouldTriggerCookiePartitioning() {
+  var userAgent =  window.navigator.userAgent;
+  var isSafari = userAgent.indexOf('Safari');
+  var versionNumber = isSafari ? parseFloat(userAgent.match(/Version\/(\d+\.?\d*)/)[1]) : null;
+  
+  // TODO: Replace with library for checking user-agents
+  return document.hasStorageAccess && isSafari && versionNumber >= 12;
```

Comment:
What about on Mobile?
POS: `com.jadedpixel.pos Shopify POS/4.8.3 (iPhone; iOS 11.4; Scale/3.00) MobileMiddlewareSupported`
Shopify Mobile:`Shopify Mobile/iOS/7.6.0 (iPhone10,5 Simulator/com.shopify.ShopifyInternal/11.4.0)`
Safari: `Mozilla/5.0 (iPhone; CPU iPhone OS 12_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.0 Mobile/15E148 Safari/604.1`

Do you have a way to test the behaviour on an iOS 12 device? Ive thus far been unable to see auth broken on my device running iOS 12 b6.

cc. @andrewapperley

Reply:
@davidmuzi I've also not seen apps broken. I'm running the latest iOS 12 beta as well. If we're using UserAgent checks then we would need to check for `iOS 12` as well. I'll reply here if I see any announcement from Apple about ITP 2.0 on iOS or if its in the release notes of a new beta.

Reply:
I agree that hasStorageAccess should be sufficient, otherwise this may miss other webkit based browsers that implement `hasStorageAccess`. Since storage access is baked right into webkit I expect other browsers will be affected over time. The list of webkit-based browsers is long, including Chrome on iOS according to this: https://en.wikipedia.org/wiki/List_of_web_browsers#WebKit-based